nRF5340 with nRF21540 EK support in the Direct Test Mode and the Radio Test (#3927)
nrfconnect/sdk-nrf: doc/nrf/includes/sample_dtm_radio_test_fem.txt

You can add support for the nRF21540 front-end module (FEM) to the sample.

> Set ``-DNRF5340EK_FEM=y`` when you invoke ``west build`` or ``cmake`` in your sample directory.

Author: Fixed

   The devicetree for the nRF21540 DK already contains the required FEM configuration, so you do not need to set an additional build option.

Sometimes, you may need to add an extra overlay file.
The easiest way to add the file when building is to set it in the ``DTC_OVERLAY_FILE`` variable.
However, that will override the default settings.
When you set the ``DTC_OVERLAY_FILE`` variable, you must specify all overlay files that are needed for building.

Follow the instructions in :ref:`cmake_options` to specify the ``DTC_OVERLAY_FILE`` variable.
For example, to build the sample from the command line for an nRF5340 DK with an attached nRF21540 EK, invoke the following command within the sample directory:
``west build -b nrf52833dk_nrf52833 -- -DDTC_OVERLAY_FILE="boards/nrf5340dk_nrf5340_cpunet.overlay;../configuration/dts-nrf21540-fem.overlay;your_overlay_file.overlay"``

See :ref:`zephyr:set-devicetree-overlays` for different ways of adding the overlay file.

Alternatively, you can copy the contents of :file:`dts-nrf21540-fem.overlay` to the board's overlay file.

.. note::
   The nRF5340 DK Network Core peripherals, like UART and SPI, share an ID and a base address.

>    To configure the nRF21540 front-end module gain, write the gain value over the SPI. In samples,

Reviewer: Every new sentence on a new line.  -> The one starting "In samples, ... needs to start a new line.

Author: Done

> 

Reviewer: Remove the blank line.

Author: Removed

> You can add support for the nRF21540 front-end module to the sample.

Reviewer: ```suggestion
You can add support for the nRF21540 front-end module (FEM) to the sample.
```

Author: Done

Set ``-DNRF5340EK_FEM=y`` when you invoke ``west build`` or ``cmake`` in your sample directory.
Follow the instructions in :ref:`cmake_options`.
For example:

``west build -b nrf5340dk_nrf5340_cpunet -- -DNRF21540EK_FEM=y``

When you set ``NRF5340EK_FEM=y`` the :file:`dts-nrf21540-fem.overlay` devicetree overlay file is used to override the default settings.
The configuration from this file is merged with the additional development kit overlay file from the :file:`boards` folder if it exists.

The :file:`dts-nrf21540-fem.overlay` is located in the |fem_file_path| folder.
Make sure that the GPIOs and the SPI configuration in the file correspond to those in which your front-end module is connected.

.. note::
   You must set ``NRF5340EK_FEM=y`` if you use the nRF21540 EK.
   If you use the nRF21540 DK, build your application for the :ref:`nrf21540dk_nrf52840 <zephyr:nrf21540dk_nrf52840>` board.

>    The devicetree for the nRF21540 DK already contains the required FEM configuration, so you do not need to set additional build option.

Reviewer: ```suggestion
   The devicetree for the nRF21540 DK already contains the required FEM configuration, so you do not need to set an additional build option.
```

Author: Done

> Sometimes you may need to add additional overlay file.

Reviewer: ```suggestion
Sometimes, you may need to add an overlay file.
```
Should we elaborate what "sometimes" mean. In which situations?
If the needed overlay file is something on top of an existing one, you could say "extra" instead of additional. (Because add additional does not really sound good)

Author: Done, I added the "extra" instead of additional. Sometimes" I mean here that it depends on the user needs but maybe this word is no needed here. What do you think ?

Reviewer: You might want to give an example case when the user would need to add the overlay file.

Author: What about: "Your application may need to add an extra overlay file". Example case is hard to invent because it only depends on the user needs some user can have need to add such file

The easiest way to add the file when building is to set it in the ``DTC_OVERLAY_FILE`` variable.

> However, doing so will override the default settings.

Reviewer: ```suggestion
However, that will override the default settings.
```

Author: Done

Follow the instructions in :ref:`cmake_options` to specify the ``DTC_OVERLAY_FILE`` variable.
For example, to build the sample from the command line for an nRF5340 DK with an attached nRF21540 EK, invoke the following command within the sample directory:

> ``west build -b nrf52833dk_nrf52833 -- -DDTC_OVERLAY_FILE="boards/nrf5340dk_nrf5340_cpunet.overlay;../configuration/dts-nrf21540-fem.overlay;your_overlay_file.overlay"``

Author: It should be. Done

``west build -b nrf52833dk_nrf52833 -- -DDTC_OVERLAY_FILE="boards/nrf5340dk_nrf5340_cpunet.overlay;../configuration/dts-nrf21540-fem.overlay;your_overlay_file.overlay"``

See :ref:`zephyr:set-devicetree-overlays` for different ways of adding the overlay file.

Alternatively, you can copy the contents of :file:`dts-nrf21540-fem.overlay` to the board's overlay file.

.. note::

>    The nRF5340 DK Network Core peripheral like UART and SPI sharing an ID and base address.

Reviewer: ```suggestion
   The nRF5340 DK Network Core peripherals, like UART and SPI, share an ID and a base address.
```

Author: Done

>    The nRF5340 DK Network Core peripherals, like UART and SPI, share an ID and a base address.

Reviewer: ```suggestion
   The nRF5340 DK network core peripherals, like UART and SPI, share an ID and a base address.
```

Author: Done

>    The nRF21540 front-end-module gain is configured by writing the gain value over SPI. In samples
>    the UART is used as a control interface or shell transport, to send the gain value, UART is temporary disabled and restarted after the SPI transfer.

Reviewer: ```suggestion
   To configure the nRF21540 front-end module gain, write the gain value over the SPI. In samples,
   UART is used as a control interface or shell transport, to send the gain value. UART is temporary disabled and restarted after the SPI transfer.
```

Author: Done

Reviewer: Still a bit suspicious with this. Does it make sense, if we remove the comma between transport and to?
